### PR TITLE
Entrust: Improve organization handling 

### PR DESCRIPTION
**Pull request by sirferl**
Entrust does not regard the organization in the CSR but needs the information send in the Additional Data sent with the request. The plugin now get's the organizations from Entrust and looks up the ID of the wanted organization from the "Organization"-field. If the Organization is not valid, possible values are displayed in an error message.

### CODE REVIEW

File: lemur/plugins/lemur_entrust/plugin.py [hunk 1]
```diff
@@ -78,11 +78,35 @@ def process_options(options):
         "eku": "SERVER_AND_CLIENT_AUTH",
         "certType": product_type,
         "certExpiryDate": validity_end,
+        # "keyType": "RSA", Entrust complaining about this parameter
```
Comment (hosseinsh) [inline lemur/plugins/lemur_entrust/plugin.py:81]: the `client_id` is missing from the data.

**Review by hosseinsh — COMMENTED** [review]
I am wondering if we should move towards issuer options, and add all the CA related info as dynamic options, similar how we are doing it with the ACME plugin: 

https://github.com/Netflix/lemur/blob/95854c6f686cce7dc8b6919388fe57abfdd03fa4/lemur/plugins/lemur_acme/plugin.py#L41-L76


This would mean one issuer per Org-ID. 
depending on the number of orgs at Entrust, it might be worthwhile to manage one or two org-based issuers, and save the extra query for each cert issuance. 

On the other hand, if the number of orgs is too high, this approach might be worth it.

File: lemur/plugins/lemur_entrust/plugin.py (hunk 1 above)
Comment (sirferl) [reply lemur/plugins/lemur_entrust/plugin.py:81]: *sigh* Had problems syncing my dev-enviroment. So somehow the two lines got lost. Thank you. I repaired it.

**Comment by sirferl** [comment]
Hi, I appreciate your comment. Actually I have thought in the same direction. But we have a varying number of organizations due to M&A. So it seems better to use the check always approach. Maybe we should implement a config-switch to turn off this check when there is only one organization?

**Comment by hosseinsh** [comment]
looks like there is an Azure plugin coming in :D 
looking forward to that PR (it suppose it is added here by mistake)

**Comment by hosseinsh** [comment]
> Hi, I appreciate your comment. Actually I have thought in the same direction. But we have a varying number of organizations due to M&A. So it seems better to use the check always approach. Maybe we should implement a config-switch to turn off this check when there is only one organization?

Sounds good, we can add that later, if we notice the overhead is too high.

**Comment by hosseinsh** [comment]
this might come handy: 
https://github.com/Netflix/lemur/blob/37e58574066921f50485398d195994b1b805c216/lemur/plugins/lemur_jks/plugin.py#L42-L46

**Comment by hosseinsh** [comment]
@sirferl would be moving the Azure plugin into a separate PR?

**Comment by sirferl** [comment]
Yeah, I got a bit carried away friday afternoon working ont the azure plugin. I will clean up the mess...

**Review by hosseinsh — APPROVED** [review]
local testing looking good. 

Thanks for the PR @sirferl. Merging shortly.

**Pull request merged**